core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/customize-interaction-modal.controller.spec.ts

```diff
@@ -0,0 +1,352 @@
+
+        expect($scope.getSaveInteractionButtonTooltip()).toBe(
+          'Please add an image for the learner to click on.');
+        injectorSpy.and.callThrough();
+        expect($scope.isSaveInteractionButtonEnabled()).toBe(false);
+      });
+
+    it('should not save interaction button tooltip when form entries' +
+      ' are invalid', function() {
+      stateCustomizationArgsService.displayed = {
+        imageAndRegions: {
+          value: {
+            imagePath: 'imagepath',
+            labeledRegions: [{
+              label: 'abc'
+            }]
+          }
+        }
+      };
+      stateCustomizationArgsService.saveDisplayedValue();
+
+      $scope.form.schemaForm = {
+        $invalid: true
+      };
+      var injectorSpy = spyOn($injector, 'get');
+
+      injectorSpy.withArgs('ImageClickInputValidationService').and
+        .returnValue(imageClickInputValidationService);
+
+      expect($scope.getSaveInteractionButtonTooltip()).toBe(
+        'Some of the form entries are invalid.');
+      injectorSpy.and.callThrough();
+      expect($scope.isSaveInteractionButtonEnabled()).toBe(false);
+    });
+
+    it('should not save interaction button tooltip when there is no warning' +
+      ' message', function() {
+      stateCustomizationArgsService.displayed = {
+        imageAndRegions: {
+          value: {
+            imagePath: 'imagepath',
+            labeledRegions: [{
+              label: 'abc'
+            }]
+          }
+        }
+      };
+      stateCustomizationArgsService.saveDisplayedValue();
+      $scope.form.schemaForm = {
+        $invalid: false,
+        $valid: true
+      };
+
+      var injectorSpy = spyOn($injector, 'get');
+      injectorSpy.withArgs('ImageClickInputValidationService').and
+        .returnValue(imageClickInputValidationService);
+
+      expect($scope.getSaveInteractionButtonTooltip()).toBe('');
+      injectorSpy.and.callThrough();
+      expect($scope.isSaveInteractionButtonEnabled()).toBe(true);
+    });
+  });
+
+  describe('when state editor is not in question mode', function() {
+    beforeEach(angular.mock.inject(function($injector, $controller) {
+      var $rootScope = $injector.get('$rootScope');
+
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+
+      spyOn(stateEditorService, 'isInQuestionMode').and.returnValue(false);
+
+      $scope = $rootScope.$new();
+      $controller('CustomizeInteractionModalController', {
+        $scope: $scope,
+        $uibModalInstance: $uibModalInstance,
+        InteractionDetailsCacheService: interactionDetailsCacheService,
+        StateCustomizationArgsService: stateCustomizationArgsService,
+        StateEditorService: stateEditorService,
+        StateInteractionIdService: stateInteractionIdService
+      });
+    }));
+
+    it('should not save interaction button tooltip when there is no' +
+      ' customization  message', function() {
```

Comment:
I don't understand this it block description.

Reply:
I've changed it, ptal